### DISCUSSION TITLE
Setting up a new project via API

### DISCUSSION THREAD

**Discussion by javelinco** [cypress-io/cypress#17398] [Questions and Help]
Is there a way to setup a new project to record to in the cypress dashboard using an api instead of a UI? I'd like to be able to automatically provision new UI applications, and create a cypress key for a new project when I do so.

I've been unsuccessful finding any documentation that might help me track this down.

**Comment by jennifer-shehane** [comment]
This is not currently possible with Cypress. It requires setting up through the UI. Can you explain a bit more about your use case? Our understanding is that setting up a new project to record would be a rare event.

**Reply by javelinco** [reply]
Any time we create a new UI or micro-UI, we want to create a new project to run our cypress tests in. Our current process for creating new UIs use a provisioning application, so that our UIs are created in a consistently valid fashion. We probably create a new UI each month or so across our organization. Being able to incorporate Cypress in these applications in a consistent "infrastructure-as-code" way would be a big win. Otherwise, the process of doing so becomes a matter of hopefully following a number of complicated documented steps correctly and without error each time.
